### ISSUE TITLE
Change optional flag format?

### ISSUE THREAD

**Issue by mlschroe** [zchunk/zchunk#12]
I wonder if the current definition of optional flags are really useful. The problem is that there's just one optional flag data area for all flags, so you do not know which parts of the data to skip is a flag is not understood. Maybe something like the following would be more versatile:

```
The preface:
+===============+============+========================+==============================+
| Data checksum | Flags (ci) | Compression type (ci ) | Optional elements count (ci) |
+===============+============+========================+==============================+
+==========================+=================================+=======================+
| Optional element Id (ci) | Optional element data size (ci) | Optional element data | ... 
+==========================+=================================+=======================+
```

I.e. have a count and then multiple optional elements with an id that defines their type and a data area. That way unknown elements can be easily skipped.

**Comment by jdieter** [comment]
This makes a lot of sense.  Since we haven't defined any optional flags/element ids, no zchunk files should be using this, so let's make this change.  Just to be clear, a zchunk reader MUST NOT look for optional elements unless mandatory flag 1 is set.

**Comment by jdieter** [comment]
Can you please look at e0a5c925bc8daa1b6c79ce20fc358b8a3f0ad9f7 and see if it makes sense?  Do let me know if there's anything there that's unclear or ambiguous.

**Comment by jdieter** [comment]
This is now fully implemented in da616772d8177fe063dfc54f94b498121791bd4d.  Thanks again, Michael!
